src/nodes/lighting/LightsNode.js

```diff
@@ -107,23 +107,31 @@ class LightsNode extends Node {
 	}
 
 	/**
-	 * Overwrites the default {@link Node#customCacheKey} implementation by including the
-	 * light IDs into the cache key.
+	 * Overwrites the default {@link Node#customCacheKey} implementation by including
+	 * light data into the cache key.
 	 *
 	 * @return {number} The custom cache key.
 	 */
 	customCacheKey() {
 
-		const lightIDs = [];
+		const hashData = [];
 		const lights = this._lights;
 
 		for ( let i = 0; i < lights.length; i ++ ) {
 
-			lightIDs.push( lights[ i ].id );
+			const light = lights[ i ];
+
+			hashData.push( light.id );
+
+			if ( light.isSpotLight === true ) {
+
+				hashData.push( Number( light.map !== null ) );
```

Comment:
We may need to consider other texture data that needed to be recompiled if it changes.

Reply:
If the user sets a material property that holds a texture to `null`, the user must set `material.needsUpdate` to `true`. However, this isn't doable for lights since it would be required to set `material.needsUpdate` to `true` for all  lit materials in the scene.

Hence, the code in `LightsNode` only needs to focus on light properties.

Reply:
Hmm... my concern is the user changing from a `Texture` to a `VideoTexture` for example.

Reply:
Does a video texture require a different kind of binding? If not, switching from one texture to another shouldn't be an issue.

In any event, textures have an `id` property as well so we could store it in the hash.

Reply:
> Does a video texture require a different kind of binding? If not, switching from one texture to another shouldn't be an issue.

This will require a different gpu-program, consequently different bindings settings. `id` would make it more stable, and we can think about optimizing it later. an "id/hash" based on the type of texture would be ideal.

Reply:
Okay, I added now the ID and use `-1` if not texture is set. That should make the hash more robust.